jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -151,7 +151,11 @@ ecma_new_ecma_string_from_utf8 (const lit_utf8_byte_t *string_p, /**< utf-8 stri
                                 lit_utf8_size_t string_size) /**< string size */
 {
   JERRY_ASSERT (string_p != NULL || string_size == 0);
-  JERRY_ASSERT (lit_is_valid_cesu8_string (string_p, string_size));
+  if (!lit_is_valid_cesu8_string (string_p, string_size))
```

Comment:
We need to understand the problem. The assert says this function can only be called with a cesu-8 string. What is exactly happening here?